x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/search/action/RenderQueryResponseSerializingTests.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.search.action;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.test.AbstractWireSerializingTestCase;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import static org.elasticsearch.search.RandomSearchRequestGenerator.randomSearchSourceBuilder;
+
+public class RenderQueryResponseSerializingTests extends AbstractWireSerializingTestCase<RenderQueryAction.Response> {
+
+    @Override
+    protected Writeable.Reader<RenderQueryAction.Response> instanceReader() {
+        return RenderQueryAction.Response::new;
+    }
+
+    @Override
+    protected RenderQueryAction.Response createTestInstance() {
+        SearchSourceBuilder searchSourceBuilder = randomSearchSourceBuilder(
```

Comment:
I think the issue comes from using `nulls` as providers for some of the pieces this method needs.

Maybe you could use a simpler alternative, similar to [this one](https://github.com/elastic/elasticsearch/blob/ad63465c6430b8ae8fb6dbefa441d1cd5d24b97d/x-pack/plugin/async-search/src/test/java/org/elasticsearch/xpack/search/SubmitAsyncSearchRequestTests.java#L73).

Reply:
Thanks, I still get the same type of error when using a simpler implementation: 
```
java.lang.IllegalArgumentException: Unknown NamedWriteable category [org.elasticsearch.search.aggregations.AggregationBuilder]
```
Looking at the code if I enter nulls they just shouldn't write a lot of the options. I wonder if updating this to return `SearchRequest` would be better.

Reply:
Found the issue. [Commit](https://github.com/elastic/elasticsearch/pull/95513/commits/1b1e2ed47a8c51b5386d498872580f57fb340a87)